### DISCUSSION TITLE
[Question] Why subscribe to mqtt topic in the loop()

### DISCUSSION THREAD

**Discussion by AxelMarquez** [taligentx/dscKeybusInterface#62] [dscKeybusInterface Discussion]
Good day 😊

Looking at the MQTT examples (HomeAssistant & HomeBridge), the mqtt.subscribe function is called every time dsc.StatusChanged = true in the loop(). But mqtt.subscribe is called as well when mqtt connects/re-connects.

I would assume the subscribe is only needed when connecting/re-connecting, is there a reason to have it in the loop()?

Many thanks for your support and awesome library!

**Comment by taligentx** [comment]
Hi @AxelMarquez - the PubSubClient needs to re-subscribe to the 'dsc/Set' topic after publishing to another topic so it can listen for commands.  If you're curious there are also some examples of this in the PubSubClient library examples.
